lib/mix/test/mix/tasks/test_test.exs

```diff
@@ -770,6 +770,40 @@ defmodule Mix.Tasks.TestTest do
     end
   end
 
+  describe "--dry-run" do
```

Comment:
More test cases to add:
- [ ] `--stale` + `--only`
- [ ] `--stale` + `--exclude`
- [ ] ...?